component/componenttest/nop_exporter.go

```diff
@@ -20,17 +20,16 @@ import (
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/component/componenthelper"
 	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/consumer"
 	"go.opentelemetry.io/collector/consumer/consumertest"
 )
 
 // nopExporterFactory is factory for nopExporter.
 type nopExporterFactory struct{}
 
-var nopExporterFactoryInstance = &nopExporterFactory{}
-
 // NewNopExporterFactory returns a component.ExporterFactory that constructs nop exporters.
 func NewNopExporterFactory() component.ExporterFactory {
-	return nopExporterFactoryInstance
+	return &nopExporterFactory{}
```

Comment:
What was the impetus for this change?  I'm not sure how important it is for testing purposes, but this causes an allocation for every call to `NewNopExporterFactory()` rather than the single allocation for all invocations that the prior implementation had.

Reply:
I made this change based on a similar implementation in the consumererror package in the permanent.go file. Shouldn't a new allocation be made for every call to the `NewNopExporterFactory()` as it would return a new nop exporter? If not, I will reverse this change

Reply:
Add the reason to the PR description. Why are we making this change - should be addressed.